### PR TITLE
[Merged by Bors] - feat: add a `DilationEquiv` for scaling in normed spaces

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/AddTorsor.lean [hunk 1]
```diff
@@ -306,3 +306,37 @@ def AffineMap.ofMapMidpoint (f : P → Q) (h : ∀ x y, f (midpoint ℝ x y) = m
...
+
+section
+
+open Dilation
+
+variable {𝕜 E : Type*} [NormedDivisionRing 𝕜] [SeminormedAddCommGroup E]
+variable [Module 𝕜 E] [BoundedSMul 𝕜 E] {P : Type*} [PseudoMetricSpace P] [NormedAddTorsor E P]
+
+-- TODO: define `ContinuousAffineEquiv` and reimplement this as one of those.
+/-- Scaling by a an element `k` of the scalar ring as a `DilationEquiv` with ratio `‖k‖₊`, mapping
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/AddTorsor.lean:320]:
```suggestion
/-- Scaling by an element `k` of the scalar ring as a `DilationEquiv` with ratio `‖k‖₊`, mapping
```

File: Mathlib/Analysis/NormedSpace/AddTorsor.lean [hunk 2]
```diff
@@ -306,3 +306,37 @@ def AffineMap.ofMapMidpoint (f : P → Q) (h : ∀ x y, f (midpoint ℝ x y) = m
...
+    rw [show edist (k • x +ᵥ c) (k • y +ᵥ c) = _ from (IsometryEquiv.vaddConst c).isometry ..]
+    exact edist_smul₀ ..⟩
+
+@[simp]
+lemma DilationEquiv.smulTorsor_ratio {c : P} {k : 𝕜} (hk : k ≠ 0) {x y : E}
+    (h : dist x y ≠ 0) : ratio (smulTorsor c hk) = ‖k‖₊ :=
+  Eq.symm <| ratio_unique_of_dist_ne_zero h <| by simp [dist_eq_norm, ← smul_sub, norm_smul]
+
+@[simp]
+lemma DilationEquiv.smulTorsor_ball_ball {c : P} {k : 𝕜} (hk : k ≠ 0) :
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/AddTorsor.lean:338]: This name suggests that LHS and RHS should be swapped...

Comment (j-loreaux) [reply Mathlib/Analysis/NormedSpace/AddTorsor.lean]: swapped, and change the name to `DilationEquiv.smulTorsor_preimage_ball`, which I think is more accurate.

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**